### ISSUE TITLE
Get payed credits for members with ranks playing on your server.

### ISSUE THREAD

**Issue by FlowScript** [Minehut/Meta#863]
### Feature request

**Checklist**
- [ ✔] I have read and agree to the README (found on https://github.com/Minehut/Meta)
- [✔ ] This is not a bug report, support request or plugin request
- [✔ ] I have searched the Issue Tracker to make sure this is not a duplicate issue
- [✔ ] This issue does not contain any sensitive information 

**Platform**
Website

**Describe the feature**

Hello,

I think it would be a cool idea to add a feature where you get payed for ranked players playing on your sever.
It could also vary by their rank; for example more expensive ranks give a higher payout.
And of course the longer they play the more you get payed.

Any thoughts?

**How the feature is useful**

It makes it easier for players to earn credits, and minehut still profits from rank purchases.

---
**Use 👍 and 👎 reactions to indicate you agree or disagree with this suggestion! Feel free to discuss the idea in the comments.**

**Comment by Blocksnmore** [comment]
I don’t think this would be maintainable for minehut mainly due to the fact the market pays out in real life money. You could end up spending more on the market from that than the rank costed causing minehut to lose money.

**Comment by Checkmate128** [comment]
What if the owner of a server has a rank? What if I were to just pay someone to sit around on my server? Seems highly exploitable and unmaintainable for Minehut.

**Comment by drtshock** [comment]
Roblox does this :)

**Comment by Tringlle** [comment]
eRrRr no

**Comment by erenkarakal** [comment]
90% of the players on every top server is AFK

**Comment by FlowScript** [comment]
> What if the owner of a server has a rank? What if I were to just pay someone to sit around on my server? Seems highly exploitable and unmaintainable for Minehut.

Minehut can easily detect if a player is being active or not, or limit the amount of credits rewarded from certain players.
Also I think it's pretty obvious that minehut wouldn't reward the owner for playing on their own server. - Same with server OPs.

**Comment by Checkmate128** [comment]
> > What if the owner of a server has a rank? What if I were to just pay someone to sit around on my server? Seems highly exploitable and unmaintainable for Minehut.
> 
> Minehut can easily detect if a player is being active or not, or limit the amount of credits rewarded from certain players. Also I think it's pretty obvious that minehut wouldn't reward the owner for playing on their own server. - Same with server OPs.

How does Minehut know if a player is an owner? Also AFK/inactivity detection can be easily avoided by using a program or hacked client to randomly move in a set area. Servers could also teleport players randomly to make Minehut think they are moving and active. Also most servers don’t use vanilla permissions which is what excluding server OPs would find and exclude.

**Comment by FlowScript** [comment]
> > > What if the owner of a server has a rank? What if I were to just pay someone to sit around on my server? Seems highly exploitable and unmaintainable for Minehut.
> > 
> > 
> > Minehut can easily detect if a player is being active or not, or limit the amount of credits rewarded from certain players. Also I think it's pretty obvious that minehut wouldn't reward the owner for playing on their own server. - Same with server OPs.
> 
> How does Minehut know if a player is an owner? Also AFK/inactivity detection can be easily avoided by using a program or hacked client to randomly move in a set area. Servers could also teleport players randomly to make Minehut think they are moving and active. Also most servers don’t use vanilla permissions which is what excluding server OPs would find and exclude.

They could detect if a player is an owner by limiting this feature to only accounts that are linked to minehut's website. Then they can tell if you own the server or not. Also the AFK/inactivity detection is a little iffy but they can limit the amount of credits a player can reward to a certain server. For example: player1 was limited to only awarding 25 credits to server1 for playing for a few hours. Then player1 went on server2 and rewarded server2 with 25 credits for playing for a few hours.

**Comment by ntrogen** [comment]
Doesn't this just make free credits for gen servers?

**Comment by MrGeoTech** [comment]
Maybe give ranked players a command to donate to the server with a cool down that varies on the rank (vip=1 per month to patron=1 per week) where it gives the servers owner account a small amount of tokens (10-25)?

**Comment by drtshock** [comment]
Revisiting this, I like it but I don't think it works for ranks in their current form. The reason it makes sense for Roblox is because it pays out a % of the player's subscription. If we moved ranks to a subscription, I think this model could make sense.
